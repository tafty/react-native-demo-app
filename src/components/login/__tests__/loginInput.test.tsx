import 'react-native'

import LoginInput from '../loginInput'
import React from 'react'
import { render } from 'react-native-testing-library'

describe('LoginInput', () => {
  it('should render as insecure', () => {
    const label = 'Email address'
    const onChangeText = jest.fn()
    const onSubmitEditing = jest.fn()
    const secureTextEntry = false
    const testID = 'test-ID'
    const value = 'email'

    const { getByTestId, toJSON } = render(
      <LoginInput
        label={label}
        onChangeText={onChangeText}
        onSubmitEditing={onSubmitEditing}
        secureTextEntry={secureTextEntry}
        testID={testID}
        value={value}
      />,
    )

    expect(getByTestId(`${testID}-input`).props.placeholder).toBe(label)
    expect(getByTestId(`${testID}-input`).props.value).toBe(value)

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render as secure', () => {
    const label = 'Password'
    const onChangeText = jest.fn()
    const onSubmitEditing = jest.fn()
    const secureTextEntry = true
    const testID = 'test-ID'
    const value = 'password'

    const { getByTestId, toJSON } = render(
      <LoginInput
        label={label}
        onChangeText={onChangeText}
        onSubmitEditing={onSubmitEditing}
        secureTextEntry={secureTextEntry}
        testID={testID}
        value={value}
      />,
    )

    expect(getByTestId(`${testID}-input`).props.placeholder).toBe(label)
    expect(getByTestId(`${testID}-input`).props.value).toBe(value)

    expect(toJSON()).toMatchSnapshot()
  })
})
