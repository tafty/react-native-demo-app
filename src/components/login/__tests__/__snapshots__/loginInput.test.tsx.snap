// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginInput should render as insecure 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flex": 0,
        "flexDirection": "column",
      },
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Email address"
      rejectResponderTermination={true}
      secureTextEntry={false}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderColor": "rgba(208, 208, 208, 1)",
          "borderRadius": 4,
          "borderWidth": 1,
          "fontSize": 16,
          "paddingLeft": 16,
          "paddingRight": 46,
          "paddingVertical": 16,
        }
      }
      testID="test-ID-input"
      underlineColorAndroid="transparent"
      value="email"
    />
  </View>
</View>
`;

exports[`LoginInput should render as secure 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "flex": 0,
        "flexDirection": "column",
      },
    ]
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
          "borderColor": "rgba(208, 208, 208, 1)",
          "borderRadius": 4,
          "borderWidth": 1,
          "fontSize": 16,
          "paddingLeft": 16,
          "paddingRight": 46,
          "paddingVertical": 16,
        }
      }
      testID="test-ID-input"
      underlineColorAndroid="transparent"
      value="password"
    />
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 20,
          },
          Object {
            "color": "rgba(0, 124, 255, 1)",
            "position": "absolute",
            "right": 16,
            "top": 16,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ïŠ³
    </Text>
  </View>
</View>
`;
